woocommerce/woocommerce-android
# Filter non-purchasable and non-published products out of the selection list

Summary
==========
⚠️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2248) to be done before merging it.

Fixes issue #5678 by filtering the selectable Product List inside the Order Creation flow to exclude non-published and non-purchasable products, since they have no reason to be included inside an Order.

Screenshots
==========
| App Product section list showing non-purchasable product  | Order Creation Product list not showing the same non-purchasable product |
| ------------- | ------------- |
| ![Screenshot_20220126_102607](https://user-images.githubusercontent.com/5920403/151174857-f74b7626-459b-410f-a709-242d6819e3ba.png) | ![Screenshot_20220126_103004](https://user-images.githubusercontent.com/5920403/151173233-a688a79f-c17b-4993-a727-a75ffe99c1ef.png) |

| App Product section list showing non-published product  | Order Creation Product list not showing the same non-published product |
| ------------- | ------------- |
| ![Screenshot_20220126_103303](https://user-images.githubusercontent.com/5920403/151174873-0a4c1928-f598-4647-9180-7a6f188f1b43.png) | ![Screenshot_20220126_103316](https://user-images.githubusercontent.com/5920403/151173789-6de5ad74-d3b7-4bb4-9a5e-3d761521f219.png) |

How to Test
==========
1. Make sure you have at least one draft product and one product without price inside your store
2. Go to the Product section of the Woo app and verify that the Product list shows everything as expected, including both products mentioned at the first step
3. Go to the Order Creation form and hit `Add Product`
4. Inside the Product selection list, verify that both products don't show up on the list in any situation, including the search results.


Update release notes:
